### PR TITLE
Add_dto_extensions_network_map_server

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/main/java/org/gridsuite/network/map/dto/utils/GeneratorStartupInfos.java
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/network/map/dto/utils/GeneratorStartupInfos.java:1]: `GeneratorStartupInfos` is not an utility class: move it to a package in `dto` (_`org.gridsuite.network.map.dto.definition.extension` for example_).

File: src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java [hunk 1]
```diff
@@ -1602,7 +1602,6 @@ public void shouldReturnHvdcShuntCompensatorMapData() throws Exception {
     @Test
     public void shouldReturnVoltageLevelsAndEquipments() throws Exception {
         succeedingTestForEquipmentsInfos(NETWORK_UUID, null, "voltage-levels-equipments", null, resourceToString("/voltage-levels-equipments-map-data.json"));
-        succeedingTestForEquipmentsInfos(NETWORK_UUID, VARIANT_ID, "voltage-levels-equipments", null, resourceToString("/voltage-levels-equipments-map-data-in-variant.json"));
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java:1605]: Why did you remove this test case?

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java [hunk 2]
```diff
@@ -61,9 +61,66 @@ public static ConnectablePositionInfos toMapConnectablePosition(Identifiable<?>
         return builder.build();
     }
 
-    public static CurrentLimitsData toMapDataCurrentLimits(Branch<?> branch, Branch.Side side) {
-        Optional<CurrentLimits> limits = side == Branch.Side.ONE ? branch.getCurrentLimits1() : branch.getCurrentLimits2();
-        return limits.map(l -> toMapDataCurrentLimits(l)).orElse(null);
+    public static Optional<ActivePowerControlInfos> toActivePowerControl(Identifiable<?> identifiable) {
+        ActivePowerControlInfos.ActivePowerControlInfosBuilder builder = ActivePowerControlInfos.builder();
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java:65]: Move the builder inside the `if` block, it isn't need elsewhere.

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java:1]: To be even more explicit, maybe move extension mapper to there own utility class (ie. `ExtensionUtils`) ?

File: src/main/java/org/gridsuite/network/map/dto/mapper/VoltageLevelInfosMapper.java [hunk 3]
```diff
@@ -132,18 +128,8 @@ protected static VoltageLevelTabInfos toTabInfos(Identifiable<?> identifiable) {
...
-            Double ipMin = nullIfNan(identifiableShortCircuit.getIpMin());
-            if (ipMin != null) {
-                builder.ipMin(ipMin / 1000.);  // to get value in kA : value is in A in iidm
-            }
-            Double ipMax = nullIfNan(identifiableShortCircuit.getIpMax());
-            if (ipMax != null) {
-                builder.ipMax(ipMax / 1000.);  // to get value in kA : value is in A in iidm
-            }
-        }
+        // TODO conversion frontend to KA
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/network/map/dto/mapper/VoltageLevelInfosMapper.java:131]: this TODO is for the front, not the back

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]
Some changes to do

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java [hunk 4]
```diff
@@ -61,9 +61,66 @@ public static ConnectablePositionInfos toMapConnectablePosition(Identifiable<?>
...
+    }
+
+    public static Optional<GeneratorShortCircuitInfos> toGeneratorShortCircuit(Generator generator) {
+        GeneratorShortCircuit generatorShortCircuit = generator.getExtension(GeneratorShortCircuit.class);
+        if (generatorShortCircuit != null) {
+            GeneratorShortCircuitInfos.GeneratorShortCircuitInfosBuilder builder = GeneratorShortCircuitInfos.builder();
+            builder.transientReactance(generatorShortCircuit.getDirectTransX());
+            builder.stepUpTransformerReactance(generatorShortCircuit.getStepUpTransformerX());
+            return Optional.of(builder.build());
+        }
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java:87]: this `if` block is not cover by a test

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java [hunk 5]
```diff
@@ -61,9 +61,66 @@ public static ConnectablePositionInfos toMapConnectablePosition(Identifiable<?>
...
+    public static Optional<GeneratorStartupInfos> toGeneratorStartup(Generator generator) {
+        GeneratorStartup generatorStartup = generator.getExtension(GeneratorStartup.class);
+        if (generatorStartup != null) {
+            GeneratorStartupInfos.GeneratorStartupInfosBuilder builder = GeneratorStartupInfos.builder();
+            builder.plannedActivePowerSetPoint(nullIfNan(generatorStartup.getPlannedActivePowerSetpoint()));
+            builder.marginalCost(nullIfNan(generatorStartup.getMarginalCost()));
+            builder.plannedOutageRate(nullIfNan(generatorStartup.getPlannedOutageRate()));
+            builder.forcedOutageRate(nullIfNan(generatorStartup.getForcedOutageRate()));
+            return Optional.of(builder.build());
+        }
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java:111]: this `if` block is not cover by a test

**Review by Tristan-WorkGH — COMMENTED** [review]
@souissimai Your missing % of coverage come from here.
It's not a bug from Sonar, I've got the same result with IntelliJ coverage tool.

EDIT: I confirm there is no test case with this extension in the result
image

File: src/main/java/org/gridsuite/network/map/dto/mapper/VoltageLevelInfosMapper.java (hunk 3 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/mapper/VoltageLevelInfosMapper.java:131]: I mean that the unitt conversion on backend and it will be done on front end now ( I left the comment to explain why I changed it)

File: src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java (hunk 1 above)
Comment (souissimai) [reply src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java:1605]: will revret it

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java (hunk 4 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java:87]: I added this extension in existing test to cover it

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java (hunk 2 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java]: Done

**Comment by souissimai** [comment]
> @souissimai Your missing % of coverage come from here. It's not a bug from Sonar, I've got the same result with IntelliJ coverage tool. Probably that there wasn't test covering this case, and now that you have touch to this code, Sonar flag you.
> 
> EDIT: I confirm there is no test case with this extension in the results image


I added test for this extension

File: src/main/java/org/gridsuite/network/map/dto/utils/ActivePowerControlInfos.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package org.gridsuite.network.map.dto.utils;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/utils/ActivePowerControlInfos.java:1]: Its not an util class
Add a package `org.gridsuite.network.map.dto.definition.extension` ?

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java [hunk 7]
```diff
@@ -62,9 +62,66 @@ public static ConnectablePositionInfos toMapConnectablePosition(Identifiable<?>
         return builder.build();
     }
 
-    public static CurrentLimitsData toMapDataCurrentLimits(Branch<?> branch, Branch.Side side) {
-        Optional<CurrentLimits> limits = side == Branch.Side.ONE ? branch.getCurrentLimits1() : branch.getCurrentLimits2();
-        return limits.map(l -> toMapDataCurrentLimits(l)).orElse(null);
+    public static Optional<ActivePowerControlInfos> toActivePowerControl(Identifiable<?> identifiable) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java:65]: return activePowerControl == null ? Optional.empty() :
            Optional.of(ActivePowerControlInfos.builder()
                .activePowerControlOn(activePowerControl.isParticipate())
                .droop(activePowerControl.getDroop()).build());

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java [hunk 8]
```diff
@@ -62,9 +62,66 @@ public static ConnectablePositionInfos toMapConnectablePosition(Identifiable<?>
...
+        var activePowerControl = identifiable.getExtension(ActivePowerControl.class);
+        if (activePowerControl != null) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java:67]: Same for the others

File: src/main/java/org/gridsuite/network/map/dto/utils/ActivePowerControlInfos.java (hunk 6 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/utils/ActivePowerControlInfos.java]: Done

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java:1]: I added a package org.gridsuite.network.map.dto.definition.extension

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java (hunk 7 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java:75]: Done

File: src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java (hunk 8 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java]: Done

File: src/main/java/org/gridsuite/network/map/dto/utils/GeneratorShortCircuitInfos.java
[the remark this answers is not in the document]
Comment (souissimai) [reply src/main/java/org/gridsuite/network/map/dto/utils/GeneratorShortCircuitInfos.java:1]: done

**Review by Tristan-WorkGH — APPROVED** [review]

**Comment by SlimaneAmar** [comment]
Add licence and author for all new files
